ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -93,6 +93,14 @@
       + "'CREATE STREAM S AS ...' will create a topic 'thing-S', where as the statement "
       + "'CREATE STREAM S WITH(KAFKA_TOPIC = 'foo') AS ...' will create a topic 'foo'.";
 
+  public static final String KSQL_SINK_TOPIC_PROPERTIES_LEGACY_CONFIG =
+      KSQL_CONFIG_PROPERTY_PREFIX + "sink.legacy.properties";
```

Comment:
I would rather call this something like `sink.topic.static.default.properties`. It expresses the semantics more cleanly. `legacy` doesn't mean much over time.

Reply:
I think we should consider removing `KSQL_SINK_TOPIC_PROPERTIES_LEGACY_CONFIG` and instead moving both `SINK_NUMBER_OF_PARTITIONS_PROPERTY` and `SINK_NUMBER_OF_REPLICAS_PROPERTY` to be `CompatibilityBreakingConfigDef` instances.

Going forward, we don't really want people setting either of these properties in their config or in their `SET` statements, i.e. we're not looking to support "static counts" as a feature. We want them to age out over time.  Hence moving them out of the main config and into `CompatibilityBreakingConfigDef`.  This also avoids adding a _new_ config to manage something we'd like to get rid of.